pkg/reconciler/broker/resources/names_test.go

```diff
@@ -153,3 +212,46 @@ func trigger(ns, n, uid string) *brokerv1beta1.Trigger {
 		},
 	}
 }
+
+func checkPanic(t *testing.T, f func()) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
+		}
+	}()
+	f()
+}
+
+func Test_hashedTruncate(t *testing.T) {
+	want := "nnn" + maxNameHash
+	if got := hashedTruncate(maxName, 35); got != want {
+		t.Errorf("Expected %q, got %q", want, got)
+	}
+}
+
+func Test_hashedTruncatePanic(t *testing.T) {
+	// panic when string is longer than max
+	checkPanic(t, func() { hashedTruncate("abcde", 4) })
+
+	// panic when string is shorter than max
+	checkPanic(t, func() { hashedTruncate("", 31) })
+
+	// no panic when max == 32
+	hashedTruncate("", 32)
+}
+
+func genString(p string, l int) string {
```

Comment:
Can we use [`strings.Repeat`](https://golang.org/pkg/strings/#Repeat) instead?

Reply:
Good idea; didn't know about that one.